metaflow/plugins/kfp/kfp.py

```diff
@@ -745,16 +753,23 @@ def pipeline_transform(op: ContainerOp):
             # Disable caching because Metaflow doesn't have memoization
             if isinstance(op, ContainerOp):
                 op.execution_options.caching_strategy.max_cache_staleness = "P0D"
-                op.container.add_env_variable(
-                    V1EnvVar(
-                        name="ARGO_NODE_NAME",
-                        value_from=V1EnvVarSource(
-                            field_ref=V1ObjectFieldSelector(
-                                field_path="metadata.annotations['workflows.argoproj.io/node-name']"
-                            )
-                        ),
+                env_vars = {
+                    "MF_POD_NAME": "metadata.name",
+                    "MF_POD_NAMESPACE": "metadata.namespace",
+                    "MF_ARGO_NODE_NAME": "metadata.annotations['workflows.argoproj.io/node-name']",
+                    "MF_ARGO_WORKFLOW_NAME": "metadata.labels['workflows.argoproj.io/workflow']",
+                    "ZODIAC_SERVICE": "metadata.labels['zodiac.zillowgroup.net/service']",
+                    "ZODIAC_TEAM": "metadata.labels['zodiac.zillowgroup.net/team']",
```

Comment:
Can we optionally include the `ZODIAC_PRODUCT` in this too?

Also guessing the decision to make this fork ZG specific is being implicitly made with this change? I think everything (largely) we've done up to this point is designed to be generic but this is obviously very ZG specific 😅

Reply:
We aren't using ZODIAC_PRODUCT yet, when we do, it'll be easy to add as a tag.